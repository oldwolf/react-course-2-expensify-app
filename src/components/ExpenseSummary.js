import React from 'react';
import { connect }  from  'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import pluralize from 'pluralize';

export const ExpenseSummary = (props) => (
  <div>
    <h1>Viewing {props.expenses.length} {pluralize('expense', props.expenses.length)} totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}</h1>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
