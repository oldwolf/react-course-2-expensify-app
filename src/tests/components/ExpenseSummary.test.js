import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList without expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
