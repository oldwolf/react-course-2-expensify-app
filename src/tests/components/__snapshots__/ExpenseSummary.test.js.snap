// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList without expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpenseSummary with expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
  </h1>
</div>
`;
